tc_core/src/http/client/client_cli.rs

```diff
@@ -4,14 +4,9 @@ use http_cache_reqwest::{Cache, CacheMode, HttpCache, HttpCacheOptions, MokaMana
 use reqwest::Client;
 use reqwest_middleware::{ClientBuilder, ClientWithMiddleware};
 
-use super::Response;
-use crate::config::{self, Upstream};
-
-#[async_trait::async_trait]
-pub trait HttpClient: Sync + Send {
-  async fn execute(&self, req: reqwest::Request) -> anyhow::Result<Response>;
-  async fn execute_raw(&self, req: reqwest::Request) -> anyhow::Result<reqwest::Response>;
-}
+use super::client::HttpClient;
```

Comment:
this needs to go to tc-cli

Reply:
It's used in `request_context` and `server_context`. So I was thinking of separating these based on architecture

like:

```rust
pub mod client;
#[cfg(not(target_arch = "wasm_32"))]
mod client_cli;
#[cfg(target_arch = "wasm_32")]
mod client_wasm;
``` 

So that methods remains same and would be easy to implement, modify and add